sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/ExtendedOpenTelemetrySdkBuilder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.incubator.config.ConfigProvider;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.OpenTelemetrySdkBuilder;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.SdkConfigProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.SdkLoggerProviderBuilder;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import java.io.Closeable;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/** An builder for creating an {@link ExtendedOpenTelemetrySdk} instance. */
+public final class ExtendedOpenTelemetrySdkBuilder {
+  private final SdkTracerProviderBuilder tracerProviderBuilder = SdkTracerProvider.builder();
+  private final SdkMeterProviderBuilder meterProviderBuilder = SdkMeterProvider.builder();
+  private final SdkLoggerProviderBuilder loggerProviderBuilder = SdkLoggerProvider.builder();
+  private ContextPropagators propagators = ContextPropagators.noop();
+  private SdkConfigProvider configProvider =
+      SdkConfigProvider.create(new OpenTelemetryConfigurationModel());
+  private Consumer<Closeable> closeableConsumer =
+      closeable -> {
+        // Default no-op closeable consumer
+      };
+
+  /** Sets the {@link ContextPropagators} to use. */
+  public ExtendedOpenTelemetrySdkBuilder setPropagators(ContextPropagators propagators) {
+    this.propagators = Objects.requireNonNull(propagators, "propagators must not be null");
+    return this;
+  }
+
+  /** Sets the {@link ConfigProvider} to use. */
+  public ExtendedOpenTelemetrySdkBuilder setConfigProvider(SdkConfigProvider configProvider) {
+    this.configProvider = Objects.requireNonNull(configProvider, "configProvider must not be null");
+    return this;
+  }
+
+  public ExtendedOpenTelemetrySdkBuilder setCloseableConsumer(Consumer<Closeable> configurator) {
+    this.closeableConsumer = Objects.requireNonNull(configurator, "configurator must not be null");
+    return this;
+  }
+
+  /**
+   * Applies a consumer callback to configure the TracerProvider being built for this OpenTelemetry.
+   *
+   * @param configurator A callback fleshing out tracers.
+   * @return this
+   */
+  public ExtendedOpenTelemetrySdkBuilder withTracerProvider(
```

Comment:
What do we get from these withers?

Reply:
This is from the entity PR where it's actually needed